File: python/examples/zcsv_print.py
from __future__ import annotations

import sys

from zippedcsv import ZippedCsv


def main() -> int:
    if len(sys.argv) < 3:
        print("Usage: python examples/zcsv_print.py <zip_file> <csv_id>")
        return 1

    zip_file = sys.argv[1]
    csv_id = int(sys.argv[2])

    try:
        zcsv = ZippedCsv(zip_file)
        csv_files = zcsv.GetTableNames()

        if csv_id < 0 or csv_id >= len(csv_files):
            raise Exception("Invalid CSV ID")

        csv_name = csv_files[csv_id]
        csv = zcsv.GetCsv(csv_name)
        for row in csv.GetData():
            print(", ".join(row.values()))
        return 0
    except Exception as error:
        print(f"Error: {error}")
        return 1


if __name__ == "__main__":
    raise SystemExit(main())

File: php/examples/zcsv_print.php
<?php
/**
 * Example of using ZippedCsv to print specified by id CSV file from 
 * a zipped file.
 * 
 * Usage:
 *   php zcsv_print.php <zip_file> <csv_id>
 */

require_once __DIR__ . '/../src/mc/csv.php';
require_once __DIR__ . '/../src/mc/zippedcsv.php';

use \mc\ZippedCsv;

if ($argc < 3) {
    echo "Usage: php zcsv_print.php <zip_file> <csv_id>\n";
    exit(1);
}

$zipFile = $argv[1];
$csvId = (int)$argv[2];

try {
    $zcsv = new ZippedCsv($zipFile);
    $csvFiles = $zcsv->GetTableNames();

    if ($csvId < 0 || $csvId >= count($csvFiles)) {
        throw new \Exception("Invalid CSV ID");
    }

    $csvName = $csvFiles[$csvId];
    $csv = $zcsv->GetCsv($csvName);

    foreach ($csv->GetData() as $row) {
        echo implode(', ', $row) . "\n";
    }

    $zcsv->Close();
} catch (\Exception $e) {
    echo "Error: " . $e->getMessage() . "\n";
    exit(1);
}
